### ISSUE TITLE
using hypre on a Sierra class machine, i.e., sys_type=blueos_3_ppc64le_ib_p9

### ISSUE THREAD

**Issue by pember1** [hypre-space/hypre#238]
I'm a developer on the MARBL project at LLNL.  I'm trying to get our Eulerian radiation diffusion package, which uses hypre,  to run on on a blueos_3_ppc64le_ib_p9 machine (rzansel for what it's worth) with GPUs.  We're using hypre-2.18.2. The diffusion package is running successfully on toss machines and a CPU build of the code on rzansel runs  successfully as well. 

Initially, we're using a CPU build of hypre with an otherwise GPU build of the code. Is there any reason why this shouldn't work? When running on rzansel like this for our initial test problem,hypre is reporting that 0 iterations and a 0 residual. Here's the sequence of hypre calls being used:

     CALL HYPRE_SStructMatrixGetObject(Amatrix, par_Amatrix, ierr)
     CALL HYPRE_SStructVectorGetObject(bvector, par_bvector, ierr)
     CALL HYPRE_SStructVectorGetObject(xvector, par_xvector, ierr)
     CALL HYPRE_ParCSRPCGSolve(solver, par_Amatrix, par_bvector, par_xvector, ierr)
     CALL HYPRE_ParCSRPCGGetNumIterations(solver, num_iterations, ierr)
     CALL HYPRE_ParCSRPCGGetFinalRelative(solver, final_res_norm, ierr)
     CALL HYPRE_SStructVectorGather(xvector, ierr)

This sequence is our CG_A option. The other options we have are reporting the same thing for this kind of build (gpu build, hypre on cpus). On toss machines and "blue cpu" the same code is doing the right thing.

Any idea what might be wrong? Is there some "switch" that needs to be set to use GPUs?

**Comment by liruipeng** [comment]
Hi @pember1 Please let me confirm: GPU-diffusion + CPU-hypre did not work, gave 0 iterations, on ansel. Is it correct? If so, are the matrix and the vector in GPU memory or CPU? Thanks!

**Comment by pember1** [comment]
Yes, "GPU-diffusion + CPU-hypre did not work, gave 0 iterations, on ansel"

The matrix and the vector are on the CPU

**Comment by victorapm** [comment]
Closing this issue since this system type has been retired. Please feel free to reopen it if needed.
